src/workerd/api/global-scope.c++

```diff
@@ -467,6 +467,10 @@ kj::Promise<WorkerInterface::AlarmResult> ServiceWorkerGlobalScope::runAlarm(
         }
       }
 
+      // We only want to retry against limits if it's an user error. By default let's assume it's our
+      // fault.
+      auto shouldRetryAgainstLimits = false;
+
       // We want to alert if we aren't going to count this alarm retry against limits
       if (auto desc = e.getDescription();
```

Comment:
I find the way this is structured confusing now, especially since both legs have `context.isOutputGateBroken` and we now OR together two things say that the retryCountsAgainstLimit.

I would much prefer to see the `!context.isOutputGateBroken()` be part of the calculation of the `shouldRetryAgainstLimits` variable so that the return value can look like:
```
return WorkerInterface::AlarmResult {
        .retry = true,
        .retryCountsAgainstLimit = shouldRetryAgainstLimits,
        .outcome = outcome
      };
```

(at which point you should also rename the variable `retryCountsAgainstLimit`.